wdl-analysis/src/handlers/completions.rs

```diff
@@ -0,0 +1,453 @@
+            true
+        } else if t.kind() == SyntaxKind::Ident {
+            t.prev_token()
+                .filter(|prev| !prev.kind().is_trivia())
+                .is_some_and(|prev| prev.kind() == SyntaxKind::Dot)
+        } else {
+            false
+        }
+    } else {
+        false
+    };
+
+    if is_member_access {
+        add_member_access_completions(document, &parent, &mut items)?;
+    } else {
+        let mut current = Some(parent);
+        while let Some(node) = current {
+            match node.kind() {
+                SyntaxKind::WorkflowDefinitionNode
+                | SyntaxKind::ScatterStatementNode
+                | SyntaxKind::ConditionalStatementNode => {
+                    add_keyword_completions(&WORKFLOW_ITEM_EXPECTED_SET, &mut items);
+                    if let Some(scope) = document.find_scope_by_position(offset.into()) {
+                        add_scope_completions(scope, &mut items);
+                    }
+                    add_stdlib_completions(&mut items);
+                    add_struct_completions(document, &mut items);
+                    add_namespace_completions(document, &mut items);
+                    add_callable_completions(document, &mut items);
+                    break;
+                }
+
+                SyntaxKind::TaskDefinitionNode => {
+                    add_keyword_completions(&TASK_ITEM_EXPECTED_SET, &mut items);
+                    if let Some(scope) = document.find_scope_by_position(offset.into()) {
+                        add_scope_completions(scope, &mut items);
+                    }
+                    add_stdlib_completions(&mut items);
+                    break;
+                }
+                SyntaxKind::RootNode => {
+                    add_keyword_completions(
+                        &TOP_EXPECTED_SET.union(TokenSet::new(&[Token::VersionKeyword as u8])),
+                        &mut items,
+                    );
+                    add_struct_completions(document, &mut items);
+                    add_namespace_completions(document, &mut items);
+                    break;
+                }
+                _ => current = node.parent(),
+            }
+        }
+    }
+
+    if partial_word.is_empty() {
+        return Ok(items);
+    }
+
+    let items = items
+        .into_iter()
+        .filter(|item| item.label.starts_with(&partial_word))
+        .collect();
+
+    Ok(items)
+}
+
+/// Generates completion items for WDL keywords based on the provided token set.
+///
+/// Converts raw token values to completion items with appropriate labels,
+/// kinds, and descriptions.
+fn add_keyword_completions(token_set: &TokenSet, items: &mut Vec<CompletionItem>) {
+    items.extend(token_set.iter().map(|raw| {
+        let token = Token::from_raw(raw);
+        let label = token
+            .describe()
+            .trim_start_matches("`")
+            .split("`")
+            .next()
+            .unwrap_or("")
```

Comment:
Should this panic instead? I don't think adding a empty completion item seems right here. I could be wrong, let me know.